.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -37,6 +37,7 @@
 <!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
 <!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
 <!--- If you plan to update documentation or tests in follow-up, please note -->
+- [ ] My PR title follows the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
```

Comment:
Instead of putting it in the checklist - could we just use that list to replace the `Types of changes` list above?

Also, I think it is ok to remove the "Component (s) if applicable" part above now since many of our PR doesn't fit to any of them and we are not really using that information.

Reply:
Or remove the `Types of changes` section and then add the following to the beginning of this template:

```
<!--- Thanks for sending a pull request! Please make sure to read the contribution guidelines, then fill out the blanks below before requesting a code review. -->

<!--- 
Name your Pull Request with one of the following prefix, e.g. "feat: add support for PyTorch", to indicate the type of changes proposed. This is based on the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary).

feat: (new feature for the user, not a new feature for build script)
fix: (bug fix for the user, not a fix to a build script)
docs: (changes to the documentation)
style: (formatting, missing semi colons, etc; no production code change)
refactor: (refactoring production code, eg. renaming a variable)
test: (adding missing tests, refactoring tests; no production code change)
chore: (updating grunt tasks etc; no production code change)
--->



```

Reply:
Then let's decide all possible types:

- feat: (new feature for the user, not a new feature for build script)
- fix: (bug fix for the user, not a fix to a build script)
- docs: (changes to the documentation)
- style: (formatting, missing semicolons, etc; no production code change)
- refactor: (refactoring production code, eg. renaming a variable)
- perf: (code changes that improve performance)
- test: (adding missing tests, refactoring tests; no production code change)
- chore: (updating grunt tasks etc; no production code change)
- build: (changes that affect the build system or external dependencies)
- ci: (changes to configuration files and scripts)
- revert: (reverts a previous commit)

Does the above list cover all possible changes?

Reply:
I was also chatting with @larme about having an (optional) hooks under `./dev/pre-commit` for dev to soft link it. We can enforce a `pre-receive` server-side hooks afterwards if the commit msg doesn't follow the format.

Reply:
ye if all the types are decided then I will write a `dev/githooks/pre-commit` hook. Contributors can install hooks we provided optionally (will updated DEVELOPMENT.md with instructions).

Reply:
done